rfcs/01929_tls_attestation.md

```diff
@@ -0,0 +1,258 @@
+- Version
+  - Necessary to allow protocol upgrades in the future
+
+##### Context
+
+**Context** is a parameter used to derive separate Tokens for different
+[assertion](#assertions) types provided by the firmware. Context value is
+represented by an assertion type, for which a Token is derived.
+
+##### Length
+
+Currently the Exported Keying Material **Length** is set to `16` bytes, similar
+to the Security Strength for _AES-128_ as defined by
+[NIST](https://www.keylength.com/en/4/).
+
+#### Assertions
+
+As defined by the
+[Enclave Key Exchange Protocol (EKEP)](https://asylo.dev/docs/concepts/ekep.html#assertion):
+
+| An assertion is a cryptographically verifiable statement of an identity; an identity corresponding to a given assertion is said to be asserted by that assertion |
+| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+
+The protocol supports multiple assertion types (e.g. Intel SGX ECDSA, AMD
+SEV-SNP, public key), each having a type uniquely identified by a string.
+
+#### Initial Handshake
+
+Since applications can run on various platforms, both client and server need to
+negotiate a list of assertions types they can provide and will accept. The
+negotiation is performed at the application level prior to remote attestation
+and looks as follows:
+
+1. **Client** sends to the **Server** a list of
+   [assertions](https://asylo.dev/docs/concepts/ekep.html#client_precommit-message)
+   types it will accept
+1. **Server** replies with a list of
+   [assertions](https://asylo.dev/docs/concepts/ekep.html#server_precommit-message)
+   types it will accept
+
+Currently we are using the following assertion types:
+
+- `intel_sgx_ecdsa_0_1_report`
+- `amd_sev_snp_0_1_report`
+- `vtpm_public_key`
+- `eddsa_public_key`
+
+Assertion types that represent TEE reports also include _Trusted Computing base
+(TCB)_ versions.
+
+#### Assertion Generation
+
+The _Generate Assertion_ procedure in the workflow is described as follows in
+the case of TEE-based remote attestation:
+
+1. Derive a **Token** from the TLS session
+   1. **Token** is generated via deriving Exported Keying Material from an
+      established TLS session with a specified set of parameters
+1. Put the **Token** into the **TEE report** request
+   1. This means that when TEE platform signs a **TEE report**, it also signs
+      the provided **Token**
+1. Request a **TEE report** from hardware
+1. Firmware generates and signs a **TEE report**
+   1. The signed **TEE report** also gives clients information about the TEE
+      itself (its version and etc.) and about the code that is running inside
+      the TEE
+1. Combine the hardware generated **TEE report** with a certificate signed by
+   the **Provider**’s root key
+   1. This is required for checking the validity of **TEE reports** and TEE
+      platform keys that were used to sign them
```

Comment:
This is only about TEE-related assertions, right?

Reply:
Correct, changed to optional step